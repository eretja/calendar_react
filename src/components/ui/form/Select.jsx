import React from "react";
import PropTypes from "prop-types";
import styles from "./select.scss";

/**
 * General component description in JSDoc format. Markdown is *supported*.
 */
const Select = ({ options }) => {
  const listItems = options.map(entry => (
    <option key={entry.text}>{entry.text}</option>
  ));
  return (
    <label>
      <select className={styles.select}>{listItems}</select>
    </label>
  );
};

Select.propTypes = {
  /** Description of prop "text". */

  options: PropTypes.array
};

Select.defaultProps = {
  options: [{ value: "0", text: "bitte w√§hlen" }]
};

export default Select;
