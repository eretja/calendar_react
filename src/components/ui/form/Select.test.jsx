import React from "react";
import Select from "./Select";
import { shallow } from "enzyme";
import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { mount } from "enzyme";
Enzyme.configure({ adapter: new Adapter() });
import renderer from "react-test-renderer";

describe("Test Select component", () => {
  it("should be defined", () => {
    let wrapper = shallow(<Select/>);
    expect(wrapper).toBeDefined();
  });
  it("should render correctly", () => {
    const tree = renderer.create(<Select />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
