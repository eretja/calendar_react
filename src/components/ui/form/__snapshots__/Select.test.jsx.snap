// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Select component should render correctly 1`] = `
<label>
  <select
    className="select"
  >
    <option>
      bitte w√§hlen
    </option>
  </select>
</label>
`;
